server/api/src/rsDataObjTrim.cpp

```diff
@@ -54,42 +56,76 @@ namespace
         return info;
     }
 
-    int get_time_of_expiration(const char* age_kw) {
+    auto get_time_of_expiration(const char* age_kw) -> std::uint64_t
+    {
         if (!age_kw) {
             return 0;
         }
+
+        const auto now = std::time(nullptr);
+        if (-1 == now) {
+            THROW(SYS_INTERNAL_ERR, fmt::format("{}: Error occurred getting current time.", __func__));
+        }
+
         try {
-            const auto age_in_minutes = std::atoi(age_kw);
-            if ( age_in_minutes > 0 ) {
-                const auto now = time(0);
-                const auto age_in_seconds = age_in_minutes * 60;
-                return now - age_in_seconds;
+            const auto age_in_minutes = std::stol(age_kw);
```

Comment:
Should we use `std::stoll` instead of `std::stol`?

Reply:
Yeah, I couldn't decide on a type here. Your suggestion works.

Reply:
Done